runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -3230,18 +3318,14 @@ MM_CopyForwardScheme::cleanRegion(MM_EnvironmentVLHGC *env, MM_HeapRegionDescrip
 		region->_markData._overflowFlags = newFlags;
 		/* Force our write of the overflow flags from our cache and ensure that we have no stale mark map data before we walk */
 		MM_AtomicOperations::sync();
-		UDATA *heapBase = (UDATA *)region->getLowAddress();
-		/* If it is tail filled region scan only survivor portion of the region */
-		if ((UDATA *)region->_copyForwardData._survivorBase > heapBase) {
-			heapBase = (UDATA *)region->_copyForwardData._survivorBase;
-		}
-		UDATA *heapTop = (UDATA *)region->getHighAddress();
-		MM_HeapMapIterator objectIterator = MM_HeapMapIterator(MM_GCExtensions::getExtensions(env), env->_cycleState->_markMap, heapBase, heapTop);
-
-		J9Object *object = NULL;
-		while (NULL != (object = objectIterator.nextObject())) {
-			scanObject(env, reservingContext, object, SCAN_REASON_OVERFLOWED_REGION);
-		}
+		if (region->isFreshSurvivorRegion()) {
+			cleanInRange(env, (UDATA *)region->getLowAddress(), (UDATA *)region->getHighAddress());
+		} else if (region->isSurvivorRegion()) {
+			GC_SurvivorMemoryIterator surVivorIterator(env, region, _compressedSurvivorTable);
+			while (surVivorIterator.next()) {
+				cleanInRange(env, (UDATA *)surVivorIterator.getCurrentLow(), (UDATA *)surVivorIterator.getCurrentHigh());
```

Comment:
let give more desriptive names: for those couple of methods:

cleanOverflownRegion
cleanOverflowInRange